src/Orleans.Runtime/Hosting/EndpointOptions.cs

```diff
@@ -8,25 +10,97 @@ namespace Orleans.Hosting
     public class EndpointOptions
     { 
         /// <summary>
-        /// The external IP address or host name used for clustering.
-        /// </summary>
-        public string HostNameOrIPAddress { get; set; }
-
-        /// <summary>
-        /// The IP address used for clustering. Will be inferred from <see cref="HostNameOrIPAddress"/> if this is not directly specified.
+        /// The IP address used for clustering.
         /// </summary>
         public IPAddress IPAddress { get; set; }
 
         /// <summary>
         /// The port this silo uses for silo-to-silo communication.
         /// </summary>
-        public int Port { get; set; }
+        public int SiloPort { get; set; }
 
         /// <summary>
         /// The port this silo uses for silo-to-client (gateway) communication. Specify 0 to disable gateway functionality.
         /// </summary>
         public int ProxyPort { get; set; }
 
-        //public bool BindToAny { get; set; }
```

Comment:
wouldn't a boolean like this be enough?

Reply:
No, with the proposed solution you can also bind to another port, different from the published port, if you want

Reply:
ah, ok, I was under the old assumption that we didn't want to support port re-mapping.